IdentityCore/src/util/NSDate+MSIDExtensions.m

```diff
@@ -59,4 +59,39 @@ + (NSDate *)msidDateFromTimeStamp:(NSString *)timeStamp
     return [NSDate dateWithTimeIntervalSince1970:[timeStamp doubleValue]];
 }
 
+/**
+ https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After
+ There are 2 formats of Retry-Header in the HTTP header ESTS may return:
+ Retry-After: Wed, 21 Oct 2015 07:28:00 GMT
+ Retry-After: 120
+ Either the value, we will return the date object.
+ */
++ (NSDate *)msidDateFromRetryHeader:(NSString *)retryHeaderString
+{
+    if(!retryHeaderString)
+    {
+        return nil;
+    }
+    
+    if ([retryHeaderString intValue] > 0)
+    {
+        NSDate *date = [[NSDate new] dateByAddingTimeInterval:[retryHeaderString intValue]];
+        return date;
+    }
+    else
+    {
+        static NSDateFormatter* s_dateFormatter = nil;
+        static dispatch_once_t s_dateOnce;
+        
+        dispatch_once(&s_dateOnce, ^{
+            s_dateFormatter = [[NSDateFormatter alloc] init];
+            s_dateFormatter.locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US"];
+            [s_dateFormatter setTimeZone:[NSTimeZone timeZoneWithName:@"GMT"]];
+            [s_dateFormatter setDateFormat:@"yyyy-MM-dd HH:mm:ss.SSSS"];
+        });
+        
+        return [s_dateFormatter dateFromString:retryHeaderString];
```

Comment:
nit: should we change this to return (s_dateFormatter) ? [s_dateFormatter dateFromString:retryHeaderString] : nil

Reply:
I'm not sure about the logic in your suggestion, can you please elaborate?

Reply:
s_dateFormatter could be nil, which could break dateFromString

Reply:
I don't think it can be nil as it's static and dispatch_once is synchronous.